sdk/python/kfp/compiler/pipeline_spec_builder.py

```diff
@@ -1628,6 +1631,94 @@ def write_pipeline_spec_to_file(pipeline_spec: pipeline_spec_pb2.PipelineSpec,
         with open(package_path, 'w') as yaml_file:
             yaml.dump(json_dict, yaml_file, sort_keys=True)
 
+        with open(package_path, 'r+') as f:
+            old = f.read()
+            f.seek(0)
+            f.write(yaml_comments + old)
+
     else:
         raise ValueError(
             f'The output path {package_path} should end with ".yaml".')
+
+
+def extract_comments_from_pipeline_spec(pipeline_spec: dict,
+                                        pipeline_description: str) -> str:
+    map_parameter_types = {
+        'NUMBER_INTEGER': 'int',
+        'NUMBER_DOUBLE': 'float',
+        'STRING': 'str',
+        'BOOLEAN': 'bool',
+        'LIST': 'list',
+        'STRUCT': 'dict'
+    }
+
+    def add_inputs():
+        if 'inputDefinitions' in pipeline_spec['root']:
```

Comment:
nit: What do you think about separating the information collection from the information presentation in `add_inputs` and `add_outputs` for readability [[ref](https://www.google.com/search?q=single+responsibility+principle&oq=single+respnosibil&aqs=chrome.1.69i57j0i13i131i433i512j0i13i512l3j69i64l3.2296j0j7&sourceid=chrome&ie=UTF-8)]?